src/cart/crt_internal.h

```diff
@@ -27,28 +27,37 @@
 #include "crt_swim.h"
 
 /* A wrapper around D_TRACE_DEBUG that ensures the ptr option is a RPC */
-#define RPC_TRACE(mask, rpc, fmt, ...)					\
-	do {								\
-		D_TRACE_DEBUG(mask, (rpc),				\
-			"[opc=%#x rpcid=%#lx rank:tag=%d:%d] " fmt,	\
-			(rpc)->crp_pub.cr_opc,				\
-			(rpc)->crp_req_hdr.cch_rpcid,			\
-			(rpc)->crp_pub.cr_ep.ep_rank,			\
-			(rpc)->crp_pub.cr_ep.ep_tag,			\
-			## __VA_ARGS__);				\
+#define RPC_TRACE(mask, rpc, fmt, ...)						\
+	do {									\
+		char *_module;							\
+		char *_opc;							\
+										\
+		crt_opc_decode((rpc)->crp_pub.cr_opc, &_module, &_opc);		\
```

Comment:
This code will be called a lot and crt_opc_decode is potentially expensive, you should use `D_LOG_ENABLED` to check if the logging will be used before calling this function.

Reply:
done